rpg.py

```diff
@@ -0,0 +1,179 @@
+from queue import PriorityQueue
+from typing import List, Tuple
+import random
+
+# Name input
+print("What is your name?")
+name = ''
+while name == '':
+    name = input('>')
+
+
+# start with the character class
+
+class Character:
+    # Class of characters
+    def __init__(self, name: str, hp: int, attack: int, speed: int, team: int):
+        self.name = name
+        self.hp = hp
+        self.attack = attack
+        self.speed = speed
+        self.team = team
+
+    def is_alive(self) -> bool:
+        """Returns true if the player if they are alive"""
+        return self.hp > 0
+
+
+
+    def is_ally(self, other_player) -> bool:
+        """returns true if the player is an ally"""
+        return other_player.team == self.team
+
+    def deal_damage(self, other_player):
+        """Deals damage if the other player is alive"""
+        if other_player.is_alive():
+            other_player.hp -= self.attack
+            print(f'{self.name} attacked {other_player.name} for {self.attack} damage')
+        return other_player
+    
+    def get_all_enemies(self, players: List) -> List[int]:
+        return [
+            index for index, player in enumerate(players)
+                if not self.is_ally(player) and player.is_alive()
+        ]
+
+    def act(self, players: List) -> List:
+        all_enemy_locations = self.get_all_enemies(players)
+        if all_enemy_locations:
+            if self.is_alive:
+                print(f'{self.name}\'s turn!')
+            targeted_index = all_enemy_locations[0]
+            targeted_player = players[targeted_index]
+            damaged_player = self.deal_damage(targeted_player)
+            players[targeted_index] = damaged_player
+            return players
+        else:
+            return players
+
+
+
+
+class Ally(Character):
+    # subclass of character
+    """This is a subclass of characters specific to team 1"""
+    def __init__(self, name: str, hp: int, attack: int, speed: int):
+        super().__init__(name, hp, attack, speed, team=1)
+
+    def act(self, players):
+        """the act method specific to team 1"""
+        # I want to input the name of the enemy to attack
+        all_enemy_locations = self.get_all_enemies(players)
+        enemies = {character.name:
+                   character for character in players if character.team != 1}
+        if all_enemy_locations:
+            if self.is_alive:
+                print(f'{self.name}\'s turn!')
+                print('Who do you attack?')
+                for player in players:
+                    if not player.is_ally(self) and player.is_alive():
+                        print(player.name)
+                targeted_enemy = None
+                while targeted_enemy == None:
+                    user_input = input('>')
+                    targeted_enemy = enemies.get(user_input)
+                    damaged_player = self.deal_damage(targeted_enemy)
+                    return players
+                else:
```

Comment:
This line of code will never be executed, since your method ends in the while loop. Remove it

Reply:
I figured that out, I had to send, Damaged player back an indent and it executes now